File: app/src/main/java/remix/myplayer/appshortcuts/AppShortcutActivity.java
package remix.myplayer.appshortcuts;

import android.content.Intent;
import android.os.Bundle;

import remix.myplayer.service.MusicService;
import remix.myplayer.ui.activity.base.BaseMusicActivity;

/**
 * Created by Remix on 2017/11/1.
 */

public class AppShortcutActivity extends BaseMusicActivity {
    public static final int SHORTCUT_TYPE_SHUFFLE_ALL = 0;
    public static final int SHORTCUT_TYPE_MY_LOVE = 1;
    public static final int SHORTCUT_TYPE_LAST_ADDED = 2;
    public static final int SHORTCUT_TYPE_CONTINUE_PLAY = 3;

    public static final String KEY_SHORTCUT_TYPE = "com.remix.myplayer.appshortcuts.ShortcutType";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int type = getIntent() != null ? getIntent().getIntExtra(KEY_SHORTCUT_TYPE,-1) : -1;
        startService(type);
        finish();
    }

    private void startService(int type){
        Intent intent = new Intent(this, MusicService.class);
        switch (type){
            case SHORTCUT_TYPE_LAST_ADDED:
                intent.setAction(MusicService.ACTION_SHORTCUT_LASTADDED);
                break;
            case SHORTCUT_TYPE_SHUFFLE_ALL:
                intent.setAction(MusicService.ACTION_SHORTCUT_SHUFFLE);
                break;
            case SHORTCUT_TYPE_MY_LOVE:
                intent.setAction(MusicService.ACTION_SHORTCUT_MYLOVE);
                break;
            case SHORTCUT_TYPE_CONTINUE_PLAY:
                intent.setAction(MusicService.ACTION_SHORTCUT_CONTINUE_PLAY);
                break;
        }
        startService(intent);
    }
}

File: app/src/main/java/remix/myplayer/appshortcuts/AppShortcutActivity.kt
package remix.myplayer.appshortcuts

import android.content.Intent
import android.os.Bundle
import remix.myplayer.service.MusicService
import remix.myplayer.ui.activity.base.BaseMusicActivity

/**
 * Created by Remix on 2017/11/1.
 */

class AppShortcutActivity : BaseMusicActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    val type = if (intent != null) intent.getIntExtra(KEY_SHORTCUT_TYPE, -1) else -1
    startService(type)
    finish()
  }

  private fun startService(type: Int) {
    val intent = Intent(this, MusicService::class.java)
    when (type) {
      SHORTCUT_TYPE_LAST_ADDED -> intent.action = MusicService.ACTION_SHORTCUT_LASTADDED
      SHORTCUT_TYPE_SHUFFLE_ALL -> intent.action = MusicService.ACTION_SHORTCUT_SHUFFLE
      SHORTCUT_TYPE_MY_LOVE -> intent.action = MusicService.ACTION_SHORTCUT_MYLOVE
    }
    startService(intent)
  }

  companion object {
    const val SHORTCUT_TYPE_SHUFFLE_ALL = 0
    const val SHORTCUT_TYPE_MY_LOVE = 1
    const val SHORTCUT_TYPE_LAST_ADDED = 2

    const val KEY_SHORTCUT_TYPE = "com.remix.myplayer.appshortcuts.ShortcutType"
  }
}
